### ISSUE TITLE
scratch scope is broken?

### ISSUE THREAD

**Issue by navytux** [iovisor/bpftrace#407]
Hello up there. If I try to use the following program:

```c
kprobe::vfs_open {
        $h = 0;
}

kretprobe::vfs_open {
        $h = 1;
}
```

(note `$h` is used in _both_ probes), bpftrace gives the following unhelpful message (hard to diagnose what is going in a real program):

```
root@deco:# bpftrace test.bt 
error: <unknown>:0:0: in function kretprobe:vfs_open i64 (i8*): A call to built-in function 'abort' is not supported.
```

However if I change `$h` in the second probe to e.g. `$g` - it works:

```c
kprobe::vfs_open {
        $h = 0;
}

kretprobe::vfs_open {
        $g = 1;
}
```

```
root@deco# bpftrace test.bt 
Attaching 2 probes...
^C
```

So it seems scope handling for scratch variables is not correct and it overlaps between probes in unexpected way.

Thanks beforehand,
Kirill

**Comment by brendangregg** [comment]
Thanks, this is #380, closing. Yes, we need to fix #380!

**Comment by navytux** [comment]
Thanks for feedback.
